ts/src/sourcemapper/sourcemapper.ts

```diff
@@ -0,0 +1,230 @@
+   * a .map file and an output file associated with it.
+   *
+   * If there is no such mapping, it could be because the input file is not
+   * the input to a transpilation process or it is the input to a transpilation
+   * process but its corresponding .map file was not given to the constructor
+   * of this mapper.
+   *
+   * @param {string} inputPath The path to an input file that could
+   *  possibly be the input to a transpilation process.  The path should be
+   *  relative to the process's current working directory.
+   */
+  hasMappingInfo(inputPath: string): boolean {
+    return this.getMappingInfo(inputPath) !== null;
+  }
+
+  /**
+   * @param {string} inputPath The path to an input file that could possibly
+   *  be the input to a transpilation process.  The path should be relative to
+   *  the process's current working directory
+   * @param {number} The line number in the input file where the line number is
+   *   zero-based.
+   * @param {number} (Optional) The column number in the line of the file
+   *   specified where the column number is zero-based.
+   * @return {Object} The object returned has a "file" attribute for the
+   *   path of the output file associated with the given input file (where the
+   *   path is relative to the process's current working directory),
+   *   a "line" attribute of the line number in the output file associated with
+   *   the given line number for the input file, and an optional "column" number
+   *   of the column number of the output file associated with the given file
+   *   and line information.
+   *
+   *   If the given input file does not have mapping information associated
+   *   with it then the input location is returned.
+   */
+  mappingInfo(location: GeneratedLocation): SourceLocation {
+    const inputPath = path.normalize(location.file);
+    const entry = this.getMappingInfo(inputPath);
+    if (entry === null) {
+      return location;
+    }
+
+    const generatedPos = {line: location.line, column: location.column};
+
+    // TODO: Determine how to remove the explicit cast here.
+    const consumer: sourceMap.SourceMapConsumer =
+        entry.mapConsumer as {} as sourceMap.SourceMapConsumer;
+
+    const pos = consumer.originalPositionFor(generatedPos);
+    if (pos.source === null) {
+      return location;
+    }
+    return {
+      file: pos.source,
+      line: pos.line || undefined,
+      name: pos.name || location.name,
+      // TODO: The `sourceMap.Position` type definition has a `column`
+      //       attribute and not a `col` attribute.  Determine if the type
+      //       definition or this code is correct.
+      column: (pos as {} as {col: number}).col
+    };
+  }
+}
+
+export async function create(sourcemapPaths: Map<string, string>):
+    Promise<SourceMapper> {
+  const limit = pLimit(CONCURRENCY);
+  const mapper = new SourceMapper();
+  const promises: Array<Promise<void>> = [];
```

Comment:
This pattern could be rewritten as a map:
```js
const promises = sourcemapPaths.map((parentDir, path) => {
   return processSourcemap(mapper.infoMap, path, parentDir);
});
```

Feel like I needed to make up for the extra few lines I caused with my lodash feedback 😆

Reply:
I get a typescript error when I try this:
```
Property 'map' does not exist on type 'Map<string, string>'.
```

Reply:
eh, either way, this is fine :)